aten/src/ATen/native/sparse/SparseTensor.cpp

```diff
@@ -284,14 +284,16 @@ SparseTensor dense_to_sparse(const Tensor& self){
 
 SparseTensor dense_to_sparse(const Tensor& self, int64_t sparse_dim){
   int64_t dims = self.dim();
-  AT_CHECK(sparse_dim > 0, "sparse_dim must be >0");
+  // TODO: it seems like sparse_dim == 0 could be supported even if self.dim() > 0,
+  // but this would take some work and doesn't seem particularly useful.
```

Comment:
ISTR myself saying that in this sense, sparse tensors are a strict generalization of dense tensors :)

Reply:
Yep, feel free to implement it to make it true!